src/Plastic.jl

```diff
@@ -48,14 +48,18 @@ end
 Base.zero(::Type{PlasticState{dim,T,M}}) where {dim,T,M} = PlasticState(zero(SymmetricTensor{2,dim,T,M}), zero(T), zero(SymmetricTensor{2,dim,T,M}), zero(T))
 initial_material_state(::Plastic) = zero(PlasticState{3,Float64,6})
 
+struct PlasticCache{T<:OnceDifferentiable} <: AbstractCache
```

Comment:
I think this could be `NLsolveCache` and live in a seperate `cache.jl` file. It is not really specific to the plastic material at all, rather applies to any material that would use nlsolve as a non-linear solver.